assets/helpers/page/page.js

```diff
@@ -0,0 +1,105 @@
+// @flow
+
+// ----- Imports ----- //
+
+import { createStore, combineReducers } from 'redux';
+import 'ophan';
+
+import * as ga from 'helpers/tracking/ga';
+import * as abTest from 'helpers/abtest';
+import * as logger from 'helpers/logger';
+import { getCampaign } from 'helpers/tracking/guTracking';
+import { getQueryParameter } from 'helpers/url';
+import { detect } from 'helpers/internationalisation/country';
+
+import type { Campaign } from 'helpers/tracking/guTracking';
+import type { IsoCountry } from 'helpers/internationalisation/country';
+import type { Participations } from 'helpers/abtest';
+
+import type { Action } from './pageActions';
+
+
+// ----- Types ----- //
+
+type CommonState = {
+  intCmp: ?string,
+  campaign: ?Campaign,
+  refpvid: ?string,
+  country: IsoCountry,
+  abParticipations: Participations,
+};
+
+
+// ----- Functions ----- //
+
+function pageInitialisation() {
+
+  // AB tests.
+  const participations = abTest.init();
+
+  // Google analytics.
+  ga.init();
+  ga.setDimension('experience', abTest.getVariantsAsString(participations));
+  ga.trackPageview();
+
+  // Logging.
+  logger.init();
+
+  return participations;
+
+}
+
+function createCommonReducer(abParticipations: Participations) {
+
+  const intCmp = getQueryParameter('INTCMP');
+
+  const initialState: CommonState = {
+    intCmp,
+    campaign: intCmp ? getCampaign(intCmp) : null,
+    refpvid: getQueryParameter('REFPVID'),
+    country: detect(),
+    abParticipations,
+  };
+
+  function commonReducer(
+    state: CommonState = initialState,
+    action: Action): CommonState {
+
+    switch (action.type) {
+
+      case 'SET_INTCMP':
+        return Object.assign({}, state, { intCmp: action.intCmp });
```

Comment:
Since this is changing intCmp, should it also set campaign based on getCampaign(intCmp) ?

Reply:
I agree. In fact, because we now have `campaign` in the redux store, we don't need to rely on `intCmp` around as much: 7c53d9a